---
title: "Interval Null Hypothesis Testing for Differential Expression and DIfferential Abundance with INDExA"
vignette: >
  %\VignetteIndexEntry{INDExA_vignette}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
author:
  - name: Kyle McGovern
    affiliation: Program in Bioinformatics and Genomics, The Pennsylvania State University
    email: kvm6065@psu.edu
output:
  BiocStyle::html_document:
    toc_float: true
  BiocStyle::pdf_document: default
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# Quick Start 1: Simple `INDExA` with Two Groups

## Setup & Code

Here we analyze simulated data from a mock experiment measuring changes in
50 microbes' abundance in the colons of 100 individuals. The individuals were
divided into two groups: 50 *untreated* individuals representing control group
and 50 *treated* individuals who ingested a selective microbial growth factor.
30 of the microbes were simulated with positive LFC between 0.25 and 2.5, and
the remaining 20 microbes were simulated with LFC of 0.

Consider, based on prior knowledge, we believe the selective microbial growth
factor could only increase the total number of microbes in the colons of
**treated** vs. **untreated** individuals by at most 3 fold (and at least 1
fold). If we represent the $log_2$ fold change in total colonic microbial
abundance as the variable $\theta^\perp$, we can represent this (correct)
interval assumption as:
$$
\begin{align*}
  \theta^\perp &\in [\log_2(1), \log_2(3)] \\
  \theta^\perp &\in [0, 1.585].
\end{align*}
$$
(**NOTE:** In this sim the true $\theta^\perp=1.12$)

Using this interval assumption, we can perform an analysis using `INDExA`
function:
```{r,message=FALSE}
library(INDExA)
data(data_sim_1)
# 0 represents untreated samples and 1 represents treated samples
conds <- c(rep(0, 50), rep(1, 50))
# Our interval assumption about the change in total microbial load
epsilon.interval <- c(0, 1.233)
# Build INDExA Object from MC Sampling
indexa.obj <- indexa.mc(data_sim_1$seq_counts, mc.samples=500)
# Run Interval Null Hypothesis Testing
indexa.sim.results <- indexa.test(indexa.obj, conds,
                                  epsilon.interval=epsilon.interval)
```

## Comparison to ALDEx2 Output

We will compare our output to the results of `ALDEx2` using CLR normalization.
We use the known LFC to determine true/false positives/negatives. Remarkably,
`INDExA` had a false positive rate of 0\% while `ALDEx2` shows a false
positive rate of about 70\%(!!!)

```{r,message=FALSE}
library(ALDEx2)
aldex2.sim.results <- aldex(data_sim_1$seq_counts,
                            conds,
                            mc.samples=500)
```

```{r,echo=FALSE}
tp_ind <- which(data_sim_1$lfc!=0)
tn_ind <- which(data_sim_1$lfc==0)
m <- rbind(c(sum(aldex2.sim.results[tp_ind,"we.eBH"]<=0.05),
             sum(aldex2.sim.results[tn_ind,"we.eBH"]>0.05),
             sum(aldex2.sim.results[tn_ind,"we.eBH"]<=0.05),
             sum(aldex2.sim.results[tp_ind,"we.eBH"]>0.05)),
           c(sum(indexa.sim.results[tp_ind,"gtt.pval.BH.adj"]<=0.05),
             sum(indexa.sim.results[tn_ind,"gtt.pval.BH.adj"]>0.05),
             sum(indexa.sim.results[tn_ind,"gtt.pval.BH.adj"]<=0.05),
             sum(indexa.sim.results[tp_ind,"gtt.pval.BH.adj"]>0.05)))
colnames(m) <- c("true positives", "true negatves",
                 "false positives", "false negatives")
row.names(m) <- c("ALDEx2", "INDExA")
m
```
